include/nlohmann/detail/output/binary_writer.hpp

```diff
@@ -861,13 +861,10 @@ class binary_writer
     */
     static std::size_t calc_bson_array_size(const typename BasicJsonType::array_t& value)
     {
-        std::size_t embedded_document_size = 0ul;
-        std::size_t array_index = 0ul;
-
-        for (const auto& el : value)
+        const std::size_t embedded_document_size = std::accumulate(std::begin(value), std::end(value), [array_index = 0ul](std::size_t result, const auto & el)
         {
-            embedded_document_size += calc_bson_element_size(std::to_string(array_index++), el);
-        }
+            return result + calc_bson_element_size(std::to_string(array_index++), el);
+        });
```

Comment:
Doesn't the lambda need to be mutable (so array_index can mutate)? If so, I'm curious how this passes unit tests; I would hope this would get checked?

Reply:
Indeed I've forgot to make it mutable !
This is weird, I run the commands describe in the [README](https://github.com/nlohmann/json#execute-unit-tests) and everything passed on my machine. At least the CI catched it

Reply:
I may have runned the tests before using the command `make amalgamate`. So this is why the tests passed on my machine before 😂 
I remove the indentifier initializer and the auto keyword in the lambda parameter since those are c++14 features, and if I understand well, this project is in c++11